### ISSUE TITLE
IDISP007 should not warn here

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#349]
```cs
namespace N
{
    using System;
    using System.Collections.ObjectModel;

    using Gu.Reactive;

    public sealed class C : IDisposable
    {
        private readonly System.Reactive.Disposables.CompositeDisposable disposable;
        private bool disposed;

        private C()
        {
            this.disposable = new System.Reactive.Disposables.CompositeDisposable
            {
                this.Xs.ObserveCollectionChangedSlim(signalInitial: false)
                       .Subscribe(_ => {}),
            };
        }

        public ObservableCollection<int> Xs { get; } = new();

        public void Dispose()
        {
            if (this.disposed)
            {
                return;
            }

            this.disposed = true;
            this.disposable?.Dispose();
        }
    }
}
```

This is a regression
